src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -1915,10 +1919,26 @@ public virtual Task<IList<TUser>> GetUsersInRoleAsync(string roleName)
         /// <param name="user"></param>
         /// <param name="methodName"></param>
         /// <returns></returns>
-        protected async Task<IdentityResult> LogResultAsync(IdentityResult result,
+        protected virtual async Task<IdentityResult> LogResultAsync(IdentityResult result,
             TUser user, [System.Runtime.CompilerServices.CallerMemberName] string methodName = "")
         {
-            result.Log(Logger, Resources.FormatLoggingResultMessage(methodName, await GetUserIdAsync(user)));
+            if (result.Succeeded)
+            {
+                if (Logger.IsEnabled(LogLevel.Information))
+                {
+                    var baseMessage = Resources.FormatLoggingResultMessage(methodName, await GetUserIdAsync(user));
+                    Logger.WriteInformation(Resources.FormatLogIdentityResultSuccess(baseMessage));
+                }
+            }
+            else
+            {
+                if (Logger.IsEnabled(LogLevel.Warning))
+                {
+                    var baseMessage = Resources.FormatLoggingResultMessage(methodName, await GetUserIdAsync(user));
+                    Logger.WriteWarning(Resources.FormatLogIdentityResultFailure(baseMessage,
+                        string.Join(",", result.Errors.Select(x => x.Code).ToList())));
+                }
+            }
             return result;
         }
 
```

Comment:
I don't really like how we have two LogResult methods here, one for result, and one for bool.  Seems pretty clunky, this might be a sideeffect of removing the virtual Log methods from the various result objects, remind me why we removed the result virtual Log methods again?